File: backend/entity/Project.go
package entity

import (
	"gorm.io/gorm"
)

type Project struct {
	gorm.Model
	Title         string    `json:"title" valid:"required~Title is required"`
	Abstract      string    `json:"abstract" valid:"required~Abstract is required"`
	Keywords      string    `json:"keywords" valid:"required~Keywords is required"`
	Year          int       `json:"year" valid:"required,range(2560|9999)~Year must be between 2560 and 99999"`
	Status        string    `json:"status" valid:"required~Status is required"`
	// FilePath      string    `json:"file_path" valid:"required~FilePath is required"`
	FilePath      string    `json:"file_path"` // Optional


	SelectionID uint          `json:"selection_id" valid:"required~SelectionID is required"`
	TopicSelection *TopicSelection `gorm:"foreignKey:SelectionID" json:"topic_selection"`
	
}

File: frontend/app/interfaces/Project.ts
export interface IProject {
    ID: number;
    title: string;
    abstract: string;
    keywords: string;
    year: number;
    status: string;
    file_path: string;
    selection_id: number;
    CreatedAt: string;
    UpdatedAt: string;
}

export interface ICreateProjectRequest {
    abstract: string;
    keywords: string;
    project_document?: File;
}

export interface IUpdateProjectRequest {
    abstract?: string;
    keywords?: string;
    status?: string;
    project_document?: File;
}
